libcraft/items/src/inventory_slot.rs

```diff
@@ -13,33 +13,37 @@ pub enum InventorySlot {
 
 impl Default for InventorySlot {
     fn default() -> Self {
-        InventorySlot::Empty
+        Self::Empty
     }
 }
 
 impl From<Option<ItemStack>> for InventorySlot {
     fn from(it: Option<ItemStack>) -> Self {
         match it {
-            Some(item) => InventorySlot::Filled(item),
-            None => InventorySlot::Empty,
+            Some(item) => Self::Filled(item),
+            None => Self::Empty,
         }
     }
 }
 
 impl InventorySlot {
+    /// Creates a new instance with the type `kind` and `count` items
+    #[must_use]
+    #[allow(clippy::missing_panics_doc)]
     pub fn new(kind: Item, count: u32) -> Self {
         if count == 0 {
             Self::Empty
         } else {
-            Self::Filled(ItemStack::new(kind, count).unwrap())
+            Self::Filled(ItemStack::new(kind, count).expect("`count` cannot equal zero"))
```

Comment:
`count` cannot be 0 here.

Reply:
Could use `ItemStack::new(kind, count).map(Self::Filed).unwrap_or_default()`.